src/components/link.js

```diff
@@ -0,0 +1,345 @@
+    highlightedColor: {default: '#24CAFF', type: 'color'},
+    visualAspectEnabled: {default: true},
+    peekMode: {default: false}
+  },
+
+  init: function () {
+    this.redirect = this.redirect.bind(this);
+    this.previousQuaternion = undefined;
+    this.hiddenEls = [];
+    this.initVisualAspect();
+  },
+
+  update: function (oldData) {
+    var data = this.data;
+    var el = this.el;
+    var strokeColor = data.highlighted ? data.highlightedColor : data.color;
+    el.setAttribute('material', 'strokeColor', strokeColor);
+    if (data.on !== oldData.on) { this.updateEventListener(); }
+    if (data.peekMode !== oldData.peekMode) { this.updatePeekMode(); }
+    if (!data.src || oldData.src === data.src) { return; }
+    el.setAttribute('material', 'pano', data.src.src);
+  },
+
+  updatePeekMode: function () {
+    var el = this.el;
+    var sphereEl = this.sphereEl;
+    if (this.data.peekMode) {
+      this.hideAll();
+      el.getObject3D('mesh').visible = false;
+      sphereEl.setAttribute('visible', true);
+    } else {
+      this.showAll();
+      el.getObject3D('mesh').visible = true;
+      sphereEl.setAttribute('visible', false);
+    }
+  },
+
+  play: function () {
+    this.updateEventListener();
+  },
+
+  pause: function () {
+    this.removeEventListener();
+  },
+
+  updateEventListener: function () {
+    var el = this.el;
+    if (!el.isPlaying) { return; }
+    this.removeEventListener();
+    el.addEventListener(this.data.on, this.redirect);
+  },
+
+  removeEventListener: function () {
+    var on = this.data.on;
+    if (!on) { return; }
+    this.el.removeEventListener(on, this.redirect);
+  },
+
+  initVisualAspect: function () {
+    var el = this.el;
+    var textEl;
+    var sphereEl;
+    var semiSphereEl;
+    if (!this.data.visualAspectEnabled) { return; }
+    textEl = this.textEl = this.textEl || document.createElement('a-entity');
+    sphereEl = this.sphereEl = this.sphereEl || document.createElement('a-entity');
+    semiSphereEl = this.semiSphereEl = this.semiSphereEl || document.createElement('a-entity');
+
+    // Set Portal
+    el.setAttribute('geometry', {primitive: 'circle', radius: 1.5, segments: 64});
+    el.setAttribute('material', {
+      shader: 'portal',
+      pano: this.data.src,
+      side: 'double'
+    });
+
+    // Set text that displays the link title / url
+    textEl.setAttribute('text', {
+      color: 'white',
+      align: 'center',
+      font: 'kelsonsans',
+      value: this.data.title || this.data.href,
+      width: 4
+    });
+    textEl.setAttribute('position', '0 1.9 0');
+    el.appendChild(textEl);
+
+    // Set the sphere that is rendered when the camera is close
+    // to the portal to allow the user peek inside
+    semiSphereEl.setAttribute('geometry', {
+      primitive: 'sphere',
+      radius: 1.5,
+      phiStart: 0,
+      segmentsWidth: 64,
+      segmentsHeight: 64,
+      phiLength: 180,
+      thetaStart: 0,
+      thetaLength: 360
+    });
+
```

Comment:
can remove a couple of these blank lines to group the entity modifications together